webapp-django/crashstats/crashstats/tests/conftest.py

```diff
@@ -0,0 +1,174 @@
+import copy
+import json
+
+import mock
+import six
+
+from django.conf import settings
+from django.contrib.auth.models import Group, Permission
+from django.contrib.contenttypes.models import ContentType
+from django.core.cache import cache
+
+from crashstats.base.tests.testbase import DjangoTestCase
+from crashstats.crashstats import models
+from crashstats.crashstats.signals import PERMISSIONS
+from crashstats.supersearch.models import SuperSearchFields
+from socorro.external.es.super_search_fields import FIELDS
+
+
+class Response(object):
+    def __init__(self, content=None, status_code=200):
+        self.raw = content
+        if not isinstance(content, six.string_types):
+            content = json.dumps(content)
+        self.content = content.strip()
+        self.status_code = status_code
+
+    @property
+    def text(self):
+        # Similar to content but with the right encoding
+        return six.text_type(self.content, 'utf-8')
+
+    def json(self):
+        return self.raw
+
+
+class ProductVersionsMixin(object):
+    """Mixin for DjangoTestCase tests to mock product versions
+
+    This mocks the function that sets the versions in the response context
+    and the versions dropdown in the navbar.
+
+    The versions dropdown is generated by get_versions_for_product which does
+    a Supersearch. This lets you mock out that function to return default
+    versions or specific versions.
+
+    Usage::
+
+        class TestViews(BaseTestViews, ProductVersionsMixin):
+            def test_something(self):
+                # ...
+                pass
+
+            def test_something_else(self):
+                self.set_product_versions(['64.0', '63.0', '62.0'])
+                # ...
+                pass
+
+    """
+    def setUp(self):
+        super(ProductVersionsMixin, self).setUp()
+        cache.clear()
+        self.mock_gvfp_patcher = mock.patch('crashstats.crashstats.utils.get_versions_for_product')
+        self.mock_gvfp = self.mock_gvfp_patcher.start()
+        self.set_product_versions(['20.0', '19.1', '19.0', '18.0'])
+
+    def tearDown(self):
+        self.mock_gvfp_patcher.stop()
+        super(ProductVersionsMixin, self).tearDown()
+
+    def set_product_versions(self, versions):
+        self.mock_gvfp.return_value = versions
+
+
+class BaseTestViews(ProductVersionsMixin, DjangoTestCase):
```

Comment:
This got moved from `crashstats.crashstats.tests.test_views` and adjusted to handle product versions.